data/backstage/infra/2025-10-30-how-the-opam-repository-works.md

```diff
@@ -0,0 +1,105 @@
+
+**Dependency Constraints**: The repository discourages overly strict version constraints (using `=` or `<=`) as they make package combination difficult and worsen the user experience. Instead, packages should use reasonable upper bounds and allow opam's solver to find compatible combinations.
+
+**Maintenance Requirements**: Packages must be actively maintained for modern OCaml versions. The current compiler cutoff threshold is OCaml 4.08, meaning packages that don't support recent compiler versions may be archived.
+
+### The Archiving Process
+
+To manage repository growth and maintain quality, the opam repository has implemented a systematic archiving process for unmaintained packages. Packages that no longer meet the compiler cutoff threshold or are explicitly marked as unmaintained get moved to the [opam-repository-archive](https://github.com/ocaml/opam-repository-archive).
+
+The process works through maintenance intent declarations using the `x-maintenance-intent` field in package metadata. When maintainers stop supporting older versions, the community is notified, and if no new maintainer steps forward, packages are archived rather than removed entirely.
+
+## The Human Element
+
+### Current Maintainer Team
+
+The repository is maintained by a dedicated team of volunteers who review every package submission. After Kate Deplaix (kit-ty-kate) retired from opam-repository maintenance in 2024 following six and a half years of service to focus on opam itself, the current active maintainers include @mseri and @raphael-proust who took over primary responsibilities. Kate's contributions were instrumental in maintaining the repository's quality and establishing many of the current processes and documentation.
+
+### Community Involvement
+
+The maintainer team actively seeks new volunteers to help manage the growing submission volume. The role offers excellent opportunities to:
+
+- Connect with and support contributors from across the OCaml ecosystem
+- Learn advanced packaging techniques and build system intricacies  
+- Contribute to ecosystem sustainability and health
+- Gain experience with large-scale open source project management
+
+New maintainers start with triaging access to review and approve changes, and after demonstrating success, can receive full access. The process includes orientation sessions for new volunteers.
+
+### Review Process
+
+Every package submission undergoes human review in addition to automated testing. Maintainers check for compliance with repository policies, verify licensing information, assess package utility, and ensure proper dependency declarations. This human oversight is crucial for maintaining the repository's curation standards.
+
+## Publishing Workflow
+
+For package authors, the typical publishing workflow involves:
+
+1. **Creating a GitHub release** of your package
```

Comment:
opam-repository pacakges are not tied to github and even for github-hosted code a release is optional

Maybe replace with
> 1. **Creating a package source archive** (this is commonly done by pushing a git tag to your online forge of choice, github or gitlab or codeberg of other)

but that's a bit verbose so maybe something simpler 

completely optional, a git tag will do, but actually even possible to do without